src/GraphQL.ApiTests/GraphQL.SystemTextJson.approved.txt

```diff
@@ -30,9 +18,40 @@ namespace GraphQL.SystemTextJson
         public static GraphQL.DI.IGraphQLBuilder AddSystemTextJson(this GraphQL.DI.IGraphQLBuilder builder, System.Action<System.Text.Json.JsonSerializerOptions, System.IServiceProvider>? action) { }
     }
-    public class InputsConverter : System.Text.Json.Serialization.JsonConverter<GraphQL.Inputs>
+    public class GraphQLRequestJsonConverter : System.Text.Json.Serialization.JsonConverter<GraphQL.Transports.Json.GraphQLRequest>
+    {
+        public GraphQLRequestJsonConverter() { }
+        public override GraphQL.Transports.Json.GraphQLRequest Read(ref System.Text.Json.Utf8JsonReader reader, System.Type typeToConvert, System.Text.Json.JsonSerializerOptions options) { }
+        public override void Write(System.Text.Json.Utf8JsonWriter writer, GraphQL.Transports.Json.GraphQLRequest value, System.Text.Json.JsonSerializerOptions options) { }
+    }
+    public class GraphQLRequestListJsonConverter : System.Text.Json.Serialization.JsonConverter<System.Collections.Generic.IEnumerable<GraphQL.Transports.Json.GraphQLRequest>>
+    {
+        public GraphQLRequestListJsonConverter() { }
+        public override bool CanConvert(System.Type typeToConvert) { }
+        public override System.Collections.Generic.IEnumerable<GraphQL.Transports.Json.GraphQLRequest> Read(ref System.Text.Json.Utf8JsonReader reader, System.Type typeToConvert, System.Text.Json.JsonSerializerOptions options) { }
+        public override void Write(System.Text.Json.Utf8JsonWriter writer, System.Collections.Generic.IEnumerable<GraphQL.Transports.Json.GraphQLRequest> value, System.Text.Json.JsonSerializerOptions options) { }
+    }
+    public class GraphQLSerializer : GraphQL.IGraphQLSerializer, GraphQL.IGraphQLTextSerializer
+    {
+        public GraphQLSerializer() { }
+        public GraphQLSerializer(GraphQL.Execution.IErrorInfoProvider errorInfoProvider) { }
+        public GraphQLSerializer(System.Action<System.Text.Json.JsonSerializerOptions> configureSerializerOptions) { }
+        public GraphQLSerializer(bool indent) { }
+        public GraphQLSerializer(System.Text.Json.JsonSerializerOptions serializerOptions) { }
+        public GraphQLSerializer(System.Action<System.Text.Json.JsonSerializerOptions> configureSerializerOptions, GraphQL.Execution.IErrorInfoProvider errorInfoProvider) { }
+        public GraphQLSerializer(bool indent, GraphQL.Execution.IErrorInfoProvider errorInfoProvider) { }
+        public GraphQLSerializer(System.Text.Json.JsonSerializerOptions serializerOptions, GraphQL.Execution.IErrorInfoProvider errorInfoProvider) { }
+        public T Deserialize<T>(string json) { }
+        public System.Threading.Tasks.ValueTask<T> ReadAsync<T>(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken = default) { }
+        public T ReadDocument<T>(System.Text.Json.JsonDocument jsonDocument) { }
+        public T ReadNode<T>(object value) { }
```

Comment:
> This particular overload could be hidden

Overload? ReadNode?

Reply:
> var request = message.PayLoad == null ? null : serializer.ReadNode((JsonElement)message.PayLoad);

Or use `var request = serializer.ReadNode(message.PayLoad);` - object version that returns default (null)

Reply:
I was saying that if we hide `ReadNode<T>(object)`, and only implement the interface member directly along with `ReadNode<T>(JsonElement)`, then you'd need to write:

```cs
OperationMessage message;

var serializer = new GraphQLSerializer();
var request = message.Payload == null ? null : serializer.ReadNode<GraphQLRequest>((JsonElement)message.Payload);

// or use the interface
IGraphQLSerializer serializer = new GraphQLSerializer();
var request = serializer.ReadNode<GraphQLRequest>(serializer);

// or cast to the interface
var serializer = new GraphQLSerializer();
var request = ((IGraphQLSerializer)serializer).ReadNode<GraphQLRequest>(serializer);
```

And that is why I left the member as public, so this code will compile:

```cs
var serializer = new GraphQLSerializer();
var request = serializer.ReadNode<GraphQLRequest>(message.Payload);
```

But since `JObject` is a class, it's not quite so much of a big deal with Newtonsoft, since this code would compile and relates what's really happening better than `ReadNode<T>(object)` does:

```cs
var serializer = new GraphQLSerializer();
var request = serializer.ReadNode<GraphQLRequest>((JObject)message.Payload);
```

Again, as it is now, `ReadNode<T>(object)` is still exposed as a public member on each serializer's class definition, along with `ReadNode<T>(JObject)` or `ReadNode<T>(JsonElement)`.

Reply:
> I was saying that if we hide ReadNode<T>(object)

I don't want to hide this one. Could we comment all specific overloads with JsonElement/JObject?

Reply:
That is, I want to say that the interface methods should be enough.